### ISSUE TITLE
Sso initiation redirects to teamcity instead of idp

### ISSUE THREAD

**Issue by chrsin** [morincer/teamcity-plugin-saml#80]
I'm experiencing a weird issue where TeamCity redirects to the url I entered in "Single Sign-on URL" but with the hostname replaced with TeamCity's own hostname.

So for an example when I enter "https://login.microsoftonline.com/guid-guid-guid-guid-guid-guid/saml2?lotsofstuffinquerystring" in Single Sign-on URL I get redirected to "https://myteamcityurl/guid-guid-guid-guid-guid-guid/saml2?lotsofstuffinquerystring" when clicking the sso button.
I've tried manually replacing the hostname of the url I get redirected to and this will initiate the login flow with the idp.

So for some reason the Single Sign On is "rewritten" to teamcitys hostname.

Prior to installing the plugin I tested it out in a docker instance on my local machine and this was **not** an issue.
The only thing I can think of is that the server on which I'm experiencing the issues is behind a reverse proxy. But I havent been able to imagine why this would cause any problems. (Also, accessing TeamCity on the server directly using the localhost url also does the redirect to the same url as I've described earlier)

Since we are never redirected to the idp, I don't feel this can be a misconfiguraiton at the idp.
I'm at my wits end here. Is there any log files I could look to for a chance to resolve this? :)

I can provide more information I'm just not sure what information to provide.

**Comment by morincer** [comment]
Hi!
Please provide the diagnosis data mentioned in section https://github.com/morincer/teamcity-plugin-saml#troubleshooting

**Comment by morincer** [comment]
Ok, well, definitely "AuthNRequest sent to https://login.microsoftonline.com" shows that the request is sent to the correct destination. 
Let's now see what happens in your browser. In Chrome, open devtools (F12), switch to the Network tab, make sure the "Preserve log" checkbox is set and re-do the sequence. Find the response from teamcity and show me the headers. I bet something is replacing the Location header

**Comment by chrsin** [comment]
These are the requests
![image](https://user-images.githubusercontent.com/365603/121664522-dc577380-caa7-11eb-8b1b-34682f0e7ba3.png)

The first request to `https://build.impact.dk/app/saml/login/` receives a 302 redirect to 
`https://build.impact.dk/671a20e3-be78-467d-8591-cc4c22259a1b/saml2?SAMLRequest=<TheFullSamlRequest>&RelayState=http%3A%2F%2Flocalhost%2Fapp%2Fsaml%2Flogin%2F`
Once the browser has been redirected it gets a 404 since TeamCity can't handle the saml request

Here is an image of the headers for the redirect
![image](https://user-images.githubusercontent.com/365603/121665925-02314800-caa9-11eb-8344-7982e35e0933.png)

**Comment by morincer** [comment]
Seems like your reverse proxy is spoiling the Location header - 302 redirect should be targeting login.microsoft.com.

**Comment by chrsin** [comment]
Am also experiencing this when accessing TeamCity directly (without going through the reverse proxy). But I just noticed that when clicking the login button I get redirected to the reverse proxy url.
I tried manually doing localhost/app/saml/login and it works (avoiding the reverse proxy)
So I think you are right. This must be the reverse proxy causing issues.

Thank you very much for your help 🥇 and so sorry I didn't catch this myself. 😞
